File: cpp/Settings/HumanCustomSkinSettings.c
// Type: Settings.HumanCustomSkinSettings
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/HumanCustomSkinSettings.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/CustomSkins/HumanCustomSkinSettings.cs
// --------------------------------

// Settings.HumanCustomSkinSettings$$.ctor
// il2cpp: void Settings_HumanCustomSkinSettings___ctor (Settings_HumanCustomSkinSettings_o* __this, const MethodInfo* method);
// 0x4085650

void Settings_HumanCustomSkinSettings___ctor(Settings_HumanCustomSkinSettings_o *__this,MethodInfo *method)

{
  Settings_BoolSetting_o *pSVar1;
  Settings_IntSetting_o *pSVar2;
  
  if (g_data_057ac3d8 == '\0') {
    il2cpp_runtime_helper_023445d0(&MethodInfo_BaseCustomSkinSettings_1_HumanCustomSkinSet);
    il2cpp_runtime_helper_023445d0(&TypeInfo_BoolSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_IntSetting);
    g_data_057ac3d8 = '\x01';
  }
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,1,(MethodInfo *)0x0);
  (__this->fields).GasEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).GasEnabled,pSVar1);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,1,(MethodInfo *)0x0);
  (__this->fields).HookEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).HookEnabled,pSVar1);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,1,(MethodInfo *)0x0);
  (__this->fields).SetSpecificSkinsEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).SetSpecificSkinsEnabled,pSVar1);
  pSVar1 = (Settings_BoolSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_BoolSetting);
  Settings_BoolSetting___ctor_40f3960(pSVar1,1,(MethodInfo *)0x0);
  (__this->fields).GlobalSkinOverridesEnabled = pSVar1;
  il2cpp_runtime_helper_022b4080(&(__this->fields).GlobalSkinOverridesEnabled);
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).SkinMode = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).SkinMode);
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).SelectedCharacterIndex = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).SelectedCharacterIndex);
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).LastGlobalPresetIndex = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).LastGlobalPresetIndex,pSVar2);
  Settings_BaseCustomSkinSettings_object____ctor((Settings_BaseCustomSkinSettings_T__o *)__this,MethodInfo_BaseCustomSkinSettings_1_HumanCustomSkinSet);
  return;
}

File: csharp_body_deref/Scripts/Settings/HumanCustomSkinSettings.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.HumanCustomSkinSettings
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/HumanCustomSkinSettings.c
// Prior real C# source (older reference): Assets/Scripts/Settings/CustomSkins/HumanCustomSkinSettings.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Settings;

[Token(Token = "0x200006B")]
internal class HumanCustomSkinSettings : BaseCustomSkinSettings<HumanCustomSkinSet>
{
	[Token(Token = "0x4000225")]
	[FieldOffset(Offset = "0x48")]
	public BoolSetting GasEnabled;

	[Token(Token = "0x4000226")]
	[FieldOffset(Offset = "0x50")]
	public BoolSetting HookEnabled;

	[Token(Token = "0x4000227")]
	[FieldOffset(Offset = "0x58")]
	public BoolSetting SetSpecificSkinsEnabled;

	[Token(Token = "0x4000228")]
	[FieldOffset(Offset = "0x60")]
	public BoolSetting GlobalSkinOverridesEnabled;

	[Token(Token = "0x4000229")]
	[FieldOffset(Offset = "0x68")]
	public IntSetting SkinMode;

	[Token(Token = "0x400022A")]
	[FieldOffset(Offset = "0x70")]
	public IntSetting SelectedCharacterIndex;

	[Token(Token = "0x400022B")]
	[FieldOffset(Offset = "0x78")]
	public IntSetting LastGlobalPresetIndex;

	[Token(Token = "0x6000310")]
	[Address(RVA = "0x3F85650", Offset = "0x3F85650", VA = "0x3F85650")]
	public HumanCustomSkinSettings()
	{
	}
}
